### DISCUSSION TITLE
How to store the historical maximum principal stress?

### DISCUSSION THREAD

**Discussion by PEI0214** [idaholab/moose#29018] [Q&A General]
### Check these boxes if you have followed the posting rules.

- [X] Q&A General is the most appropriate section for my question
- [X] I have consulted the posting Guidelines on the Discussions front page
- [X] I have searched the Discussions forum and my question has not been asked before
- [X] I have searched the MOOSE website and the documentation does not answer my question
- [X] I have formatted my post following the posting guidelines (screenshots as a last resort, triple back quotes around pasted text)

### Question

Hello，
I want to create a material model. The elastic tensor of the model is related to the historical maximum compressive stress of the large principal stress. I write a header file and a source file. Is that appropriate? Can the maximum compressive stress be stored?
[ComputeElasticityTensor.h.pdf](https://github.com/user-attachments/files/17647561/ComputeElasticityTensor.h.pdf)
[ComputeElasticityTensor.C.pdf](https://github.com/user-attachments/files/17647565/ComputeElasticityTensor.C.pdf)

**Comment by GiudGiud** [comment]
Hello

This will keep the local maximum. Do you need this or the global maximum?
My only concern here is that Cijkl is set with the old property so it's effectively lagged by a time step. Did you intend to do that?

**Reply by PEI0214** [reply]
>This will keep the local maximum. Do you need this or the global maximum?

What I need is the maximum value that occurred before the current calculation step. Can the above be done?

>My only concern here is that Cijkl is set with the old property so it's effectively lagged by a time step.  Did you intend to do that?

It needs to lag by one time step. The stress is calculated based on strain and Cijkl. When calculating Cijkl, the stress of this time step is not yet obtained, is it not?

**Reply by GiudGiud** [reply]
> What I need is the maximum value that occurred before the current calculation step. Can the above be done?

so the global spatial maximum? Or the local timestep maximum?

**Reply by PEI0214** [reply]
The maximum value of a local time step. Just before the current time step.

**Reply by GiudGiud** [reply]
local is spatial.
is this the maximum over the entire (spatial) domain, or the local maximum at the quadrature point over the history of the system

**Reply by PEI0214** [reply]
The maximum value at each quadrature point.

**Reply by GiudGiud** [reply]
ok then what you have seems good to me

**Reply by PEI0214** [reply]
Well, thank you for your help.
